tests/__snapshots__/index.test.js.snap

```diff
@@ -1763,6 +1763,137 @@ exports[`createbom produces a BOM in JSON format that includes hashes from packa
+        },
+        {
+          \\"type\\": \\"vcs\\",
+          \\"url\\": \\"git+https://github.com/d3/d3-array.git\\"
+        }
+      ]
+    },
+    {
+      \\"type\\": \\"library\\",
+      \\"bom-ref\\": \\"pkg:npm/internmap@2.0.3\\",
+      \\"name\\": \\"internmap\\",
+      \\"version\\": \\"2.0.3\\",
+      \\"description\\": \\"Map and Set with automatic key interning\\",
+      \\"hashes\\": [
+        {
+          \\"alg\\": \\"SHA-512\\",
+          \\"content\\": \\"e4787b635c106ef639a281a03db0da2f98982c03f331352b8ccba5b241cb1fac27bff03ed6ae6b8046a2baa12307ea3119721566cd4517e47f28f37765241862\\"
+        }
+      ],
+      \\"licenses\\": [
+        {
+          \\"license\\": {
+            \\"id\\": \\"ISC\\",
+            \\"text\\": {
+              \\"content\\": \\"Copyright 2021 Mike Bostock\\\\n\\\\nPermission to use, copy, modify, and/or distribute this software for any purpose\\\\nwith or without fee is hereby granted, provided that the above copyright notice\\\\nand this permission notice appear in all copies.\\\\n\\\\nTHE SOFTWARE IS PROVIDED \\\\\\"AS IS\\\\\\" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH\\\\nREGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY AND\\\\nFITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,\\\\nINDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS\\\\nOF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER\\\\nTORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF\\\\nTHIS SOFTWARE.\\\\n\\"
+            }
+          }
+        }
+      ],
+      \\"purl\\": \\"pkg:npm/internmap@2.0.3\\",
+      \\"externalReferences\\": [
+        {
+          \\"type\\": \\"website\\",
+          \\"url\\": \\"https://github.com/mbostock/internmap/\\"
+        },
+        {
+          \\"type\\": \\"issue-tracker\\",
+          \\"url\\": \\"https://github.com/mbostock/internmap/issues\\"
+        },
+        {
+          \\"type\\": \\"vcs\\",
+          \\"url\\": \\"git+https://github.com/mbostock/internmap.git\\"
+        }
+      ]
+    }
+  ]
+}"
+`;
+
+exports[`createbom produces a BOM when all dependencies are dev-dependencies that shall not be listed 1`] = `
+"{
+  \\"bomFormat\\": \\"CycloneDX\\",
+  \\"specVersion\\": \\"1.3\\",
+  \\"version\\": 1,
+  \\"metadata\\": {
+    \\"timestamp\\": \\"2020-01-01T01:00:00.000Z\\",
+    \\"tools\\": [
+      {
+        \\"vendor\\": \\"CycloneDX\\",
+        \\"name\\": \\"Node.js module\\",
+        \\"version\\": \\"3.0.0\\"
+      }
+    ],
+    \\"component\\": {
+      \\"type\\": \\"library\\",
+      \\"bom-ref\\": \\"pkg:npm/with-dev-dependencies@0.0.0\\",
+      \\"name\\": \\"with-dev-dependencies\\",
+      \\"version\\": \\"0.0.0\\",
+      \\"description\\": \\"test env to see if dev-dependencies are detected as such\\",
+      \\"purl\\": \\"pkg:npm/with-dev-dependencies@0.0.0\\"
+    }
+  },
+  \\"components\\": []
```

Comment:
no components.
all were dev-dependencies, but dev deps were not wanted to be listed.

everything as expected